src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -153,27 +149,36 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
             var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
             var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
 
-            var elementAccessExpressions = syntaxRoot.DescendantNodes().Where(syntaxFacts.IsElementAccessExpression);
             var locations = ArrayBuilder<FinderLocation>.GetInstance();
+            var indexerReferenceExpresssions = GetIndexerReferenceExpressions(syntaxRoot, syntaxFacts);
 
-            foreach (var node in elementAccessExpressions)
+            foreach (var node in indexerReferenceExpresssions)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
                 var (matched, reason) = symbolsMatch(node, semanticModel);
                 if (matched)
                 {
-                    syntaxFacts.GetPartsOfElementAccessExpression(node, out var expression, out var argumentList);
-
-                    if (symbolsMatch(expression, semanticModel).matched)
+                    SyntaxNode indexerReference;
+                    if (syntaxFacts.IsElementAccessExpression(node))
```

Comment:
you already have to specialized for ElemetnAccessExpression, so just do the same for ConditionalAccess and Cref here as well.